Sources/Gir2SwiftPlugin/plugin.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import PackagePlugin
+
+extension Path {
+    func exists() -> Bool {
+        FileManager.default.fileExists(atPath: string)
+    }
+}
+
+let module = targetBuildContext.moduleName
+let outputDir = targetBuildContext.outputDirectory.appending("Generated")
+
+let output = outputDir.appending("Output.swift")
+
+func packageDirectory(for target: Path) -> Path {
+    var curr = target
+    while !curr.appending("Package.swift").exists() {
+        if curr.stem.isEmpty {
+            fatalError("Could not find Package.swift for target \(target)")
+        }
+        curr = curr.removingLastComponent()
+    }
+    return curr
+}
+
+let girNames = try targetBuildContext.dependencies.compactMap { target -> String? in
+    let package = packageDirectory(for: target.targetDirectory)
+    let manifest = package.appending("gir2swift-manifest.sh")
+    guard manifest.exists() else { return nil }
+
+    let pipe = Pipe()
+
+    let proc = Process()
+    proc.executableURL = URL(fileURLWithPath: manifest.string)
+    proc.arguments = ["gir-name"]
+    proc.standardOutput = pipe
+    try proc.run()
+
+    return String(data: pipe.fileHandleForReading.readDataToEndOfFile(), encoding: .utf8)?
+        .trimmingCharacters(in: .whitespacesAndNewlines)
+}
+
+let girSearchPaths: [Path] = ["/opt/homebrew/share/gir-1.0", "/usr/local/share/gir-1.0", "/usr/share/gir-1.0"]
```

Comment:
I have been digging through this before (it was even accepterd) and I have found an issue that stopped me from exploring this further. It is stated in the proposal that: 

> Applying a plugin to a target is done by executing the Swift script that implements it, passing inputs that describe the target to which the plugin is being applied, and interpreting its outputs as instructions for SwiftPM. The plugin is run — either by interpreting it or by compiling it to an executable and running it —in a sandbox that prevents network access and that restricts file system access.

Was the file system restriction lifted? If not, I would suggest we find a workaround for the time being.

Other sources: 
[1] https://github.com/rhx/SwiftGtk/issues/48
[2] https://forums.swift.org/t/se-0303-package-manager-extensible-build-tools/45106/5

Reply:
I was concerned about sandboxing too, however surprisingly enough it seems like the plugin was able to access the GIR files without any issues (although admittedly I only tested this through Xcode's SPM integration, so `swift build` might behave differently).

I don't think sandboxing will be fundamentally problematic though. If the plugin itself can't access the GIR files, we could simply defer the determination of the GIR directory into gir2swift; from what I understand the command invoked by the plugin isn't as restricted as the plugin itself.

Reply:
I just checked and `swift build` works fine too – at least on macOS, which is where I tested this. Anyway, I went over the package plugins proposal again, and my suspicion seems correct: the sandbox for the executed command is indeed less restrictive than the sandbox for the plugin target.

> In addition, SwiftPM and IDEs that use libSwiftPM should run each command generated by an plugin in a sandbox. This sandbox should prevent all network access, and should only allow writing to the directories specified as outputs by the plugin.

Note that in contrast with the plugin itself (which per the spec only has access to the package dir), the sandbox for the build command is slightly less restrictive in that it has read-only access to everything, but can write to the output dir. Sandboxing the build command more strictly would probably break tons of other tools as well, so this does make sense.